runners/flink/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/ExecutableStageDoFnOperator.java

```diff
@@ -394,11 +392,16 @@ public void processWatermark(Watermark mark) throws Exception {
       } else {
         // It is not safe to advance the output watermark yet, so add a hold on the current
         // output watermark.
-        setPushedBackWatermark(Math.min(currentOutputWatermark, getPushbackWatermarkHold()));
+        long previousHold = getPushbackWatermarkHold();
```

Comment:
This still doesn't work because multiple watermarks arriving before bundle completes will wipe out the previous hold.